### PR TITLE
cdb: Delegation table emitter and flusher

### LINKED ISSUE #2086: cdb: Delegation table emitter and flusher
Track all delegations that can happen on chain
- Delegate
- Undelegate
- Redelegate

Design message between emitter and flusher first and discuss with team.

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixed: #2086  

## Implementation details
1. Add Delegations table
2. Implement emitDelegation to update message to Kafka
3. Implement handler in flusher
---

Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: chain/emitter/delegation.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+)
+
+func (app *App) emitDelegation(operatorAddress sdk.ValAddress, delegatorAddress sdk.AccAddress) {
+	fmt.Println("yo")
+	fmt.Println(operatorAddress)
+	fmt.Println(delegatorAddress)
```
Comment (tansawit) [inline chain/emitter/delegation.go]: Remove "yo"

**Review by tansawit — DISMISSED** [review]

File: flusher/flusher/db.py [hunk 2]
```diff
@@ -188,3 +188,13 @@ def Column(*args, **kwargs):
     Column("tokens", sa.Integer),
     Column("delegator_shares", sa.String),
 )
+
+delegations = sa.Table(
+    "delegations",
+    metadata,
+    Column("delegator_address", sa.String, primary_key=True),
+    Column("operator_address", sa.String, primary_key=True),
+    Column("shares", sa.String),
```
Comment (sorawit) [inline flusher/flusher/db.py]: Why is this string?

File: flusher/flusher/db.py [hunk 3]
```diff
@@ -188,3 +188,13 @@ def Column(*args, **kwargs):
     Column("tokens", sa.Integer),
     Column("delegator_shares", sa.String),
 )
+
+delegations = sa.Table(
+    "delegations",
+    metadata,
+    Column("delegator_address", sa.String, primary_key=True),
```
Comment (sorawit) [inline flusher/flusher/db.py]: Foreign key?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: flusher/flusher/handler.py [hunk 4]
```diff
@@ -73,3 +74,16 @@ def handle_update_validator(self, msg):
...
         self.conn.execute(validators.update().where(condition).values(**msg))
+
+    def handle_set_delegation(self, msg):
+        self.conn.execute(
+            insert(delegations)
+            .values(**msg)
+            .on_conflict_do_update(constraint="delegations_pkey", set_=msg)
+        )
+
+    def handle_remove_delegation(self, msg):
```
Comment (sorawit) [inline flusher/flusher/handler.py:85]: I think we can delete

File: flusher/flusher/db.py [hunk 5]
```diff
@@ -190,3 +190,18 @@ def Column(*args, **kwargs):
...
+    Column("delegator_address", sa.String, sa.ForeignKey("accounts.address"), primary_key=True),
+    Column(
+        "operator_address",
+        sa.String,
+        sa.ForeignKey("validators.operator_address"),
+        primary_key=True,
+    ),
+    Column("shares", sa.DECIMAL),
+    Column("last_ratio", sa.String),
+    Column("active", sa.Boolean),
```
Comment (sorawit) [inline flusher/flusher/db.py]: Let's not have this col

File: flusher/flusher/db.py [hunk 6]
```diff
@@ -190,3 +190,18 @@ def Column(*args, **kwargs):
...
+    metadata,
+    Column("delegator_address", sa.String, sa.ForeignKey("accounts.address"), primary_key=True),
+    Column(
+        "operator_address",
+        sa.String,
+        sa.ForeignKey("validators.operator_address"),
+        primary_key=True,
+    ),
+    Column("shares", sa.DECIMAL),
+    Column("last_ratio", sa.String),
```
Comment (sorawit) [inline flusher/flusher/db.py]: Decimal?

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

**Review by tansawit — APPROVED** [review]

**Pull request merged**